var/spack/repos/builtin/packages/kokkos-kernels/package.py

```diff
@@ -6,48 +6,261 @@
 
 
-class KokkosKernels(MakefilePackage):
+class KokkosKernels(Package):
     """Kokkos C++ Performance Portability Programming EcoSystem: Math Kernels -
     Provides BLAS, Sparse BLAS and Graph Kernels."""
 
     homepage = "https://github.com/kokkos/kokkos-kernels"
     url      = "https://github.com/kokkos/kokkos-kernels/archive/2.7.00.tar.gz"
+    git      = "https://github.com/kokkos/kokkos-kernels.git"
 
+    version('develop', branch='develop')
+    version('2.9.00', sha256='dafa9ebcdbcdc95641a986a80fcd3e532f1d80692c80a51cb08182c02c1added')
+    version('2.8.00', sha256='c9648dbf5b9cbddf2aa72f6d214a2c40e23e51660f903b83743617251574fe06')
     version('2.7.00', sha256='adf4af44eadbdfbeb9ec69dd5fae4e2852bd1fbe4a69213efd199e49f4098254')
-    version('2.6.00', sha256='14ebf806f66b9ca73949a478b8d959be7fa1165a640935760a724d7cc0a66335')
     version('2.5.00', sha256='2c2289da3a41dafd97726e90507debafbb9f5e49ca5b0f5c8d1e044a5796f000')
-    version('develop', git='https://github.com/kokkos/kokkos-kernels',
-            branch='develop')
 
-    # make sure kokkos kernels version matches kokkos
-    depends_on('kokkos@2.5.00', when='@2.5.00')
-    depends_on('kokkos@2.6.00', when='@2.6.00')
-    depends_on('kokkos@2.7.00', when='@2.7.00')
-    depends_on('kokkos@develop', when='@develop')
+    variant('debug', default=False, description="Build debug version of Kokkos")
 
-    patch('makefile.patch')
+    variant('serial', default=True, description="enable Serial backend (default)")
+    variant('pthreads', default=False, description="enable Pthreads backend")
+    variant('qthreads', default=False, description="enable Qthreads backend")
+    variant('cuda', default=False, description="enable Cuda backend")
+    variant('openmp', default=False, description="enable OpenMP backend")
 
-    def edit(self, spec, prefix):
-        makefile = FileFilter("src/Makefile")
-        makefile.filter('CXX = .*', 'CXX = ' + env['CXX'])
+    # Compilation options
+    variant('pic', default=False,
+            description="enable position independent code (-fPIC flag)")
 
-    def build(self, spec, prefix):
-        with working_dir('build', create=True):
-            makefile_path = '%s%s' % (self.stage.source_path, '/src/Makefile')
-            copy(makefile_path, 'Makefile')
-            make_args = [
-                'KOKKOSKERNELS_INSTALL_PATH=%s' % prefix,
-                'KOKKOSKERNELS_PATH=%s' % self.stage.source_path,
-                'KOKKOS_PATH=%s' % spec['kokkos'].prefix
-            ]
+    # Kokkos options
+    variant('aggressive_vectorization', default=False,
+            description="set aggressive_vectorization Kokkos option")
+    variant('disable_profiling', default=False,
+            description="set disable_profiling Kokkos option")
+    variant('disable_dualview_modify_check', default=False,
+            description="set disable_dualview_modify_check Kokkos option")
+    variant('enable_profile_load_print', default=False,
```

Comment:
All these enable/disable/use variants should have the first part removed. For example, `+disable_profiling` and `~disable_profiling` introduce a confusing double negative, but `+profiling` and `~profiling` are more clear.